### PR TITLE
build: create gh action to run benchmark

### LINKED ISSUE #6: Run the benchmarks automatically on `benchmarking-runner`
**Technical Requirements**
- Run benchmarks for all PRs on `benchmarking-runner` runner.
- It should follow the steps we have documented and automatically generate
  1. The current README.md with a table sorted highest to lowest in terms by RPS.
  2. A histogram of RPS
  3. A histogram of Latency

### PR DESCRIPTION
**Pull request by ologbonowiwi**
Example of action successfully running on GH Actions (on my fork): https://github.com/ologbonowiwi/graphql-benchmarks/actions/runs/6901170980/job/18775508812

/claim #6 
fixes #6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,14 +42,12 @@ Get started with the benchmarks:
...
-| [Gqlgen]         |     `935.00` |     `115.73` |
-| [Apollo GraphQL] |     `793.37` |     `128.22` |
-| [Netflix DGS]    |     `597.39` |     `191.85` |
-
+| Server | Requests/sec | Latency (ms) |
+|--------|--------------|--------------|
+| apollo | 1729.34 | 57.7767 |
+| netflixdgs | 1451.72 | 73.0733 |
+| gqlgen | 2095.54 | 49.4567 |
+| tailcall | 6086.87 | 16.4067 |
```
Comment (tusharmath) [inline README.md:50]: - Should be sorted in decreasing order of RPS
- Numbers should be formatted comma separated and using monospace fonts

File: .github/workflows/benchmarking-runner.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          imageName: graphql-benchmarks
+          push: never
+          runCmd: |
+            bash ./setup.sh
+            bash ./run_benchmarks.sh
+
+      - name: Push changes
+        uses: ad-m/github-push-action@master
+        with:
+          branch: ${{ github.head_ref }}
```
Comment (tusharmath) [inline .github/workflows/benchmarking-runner.yml:36]: Use the runner below to make the results are consistent.

Screenshot 2023-11-17 at 1 25 16 PM

File: README.md [hunk 3]
```diff
@@ -42,14 +42,12 @@ Get started with the benchmarks:
...
-| [Gqlgen]         |     `935.00` |     `115.73` |
-| [Apollo GraphQL] |     `793.37` |     `128.22` |
-| [Netflix DGS]    |     `597.39` |     `191.85` |
-
+| Server | Requests/sec | Latency (ms) |
+|--------|--------------|--------------|
+| [Tailcall] | `6128.85` | `16.2933` |
+| [Gqlgen] | `2101.06` | `49.8433` |
+| [Apollo GraphQL] | `1707.75` | `58.47` |
+| [Netflix DGS] | `1453.24` | `72.4867` |
```
Comment (tusharmath) [inline README.md:50]: Can you keep the numbers rounded off to two decimals

File: README.md [hunk 4]
```diff
@@ -43,12 +43,12 @@ Get started with the benchmarks:
...
-| [Tailcall]       |   `2,890.68` |      `34.69` |
-| [Gqlgen]         |     `935.00` |     `115.73` |
-| [Apollo GraphQL] |     `793.37` |     `128.22` |
-| [Netflix DGS]    |     `597.39` |     `191.85` |
+| Server | Requests/sec | Latency (ms) |
+|--------|--------------|--------------|
+| [Gqlgen] | ``2059.85`` | ``50.35`` |
+| [Apollo GraphQL] | ``1609.39`` | ``62.35`` |
+| [Netflix DGS] | ``1445.55`` | ``72.96`` |
+| [Tailcall] | ``-nan`` | ``-nan`` |
```
Comment (ologbonowiwi) [inline README.md:51]: I'm still working on this

File: README.md [hunk 5]
```diff
@@ -43,12 +43,12 @@ Get started with the benchmarks:
...
 <!-- PERFORMANCE_RESULTS_START -->
 
-| Server           | Requests/sec | Latency (ms) |
-| ---------------- | -----------: | -----------: |
-| [Tailcall]       |   `2,890.68` |      `34.69` |
-| [Gqlgen]         |     `935.00` |     `115.73` |
-| [Apollo GraphQL] |     `793.37` |     `128.22` |
-| [Netflix DGS]    |     `597.39` |     `191.85` |
+| Server | Requests/sec | Latency (ms) |
+|--------|--------------|--------------|
```
Comment (tusharmath) [inline README.md:47]: Use `:` to make sure it's right aligned.

Comment (ologbonowiwi) [reply README.md]: done

File: .github/workflows/bench.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+on: # rebuild any PRs and main branch changes
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (ologbonowiwi) [inline .github/workflows/bench.yml:11]: let's commit this right before merge it

```suggestion
    runs-on: benchmarking-runner
```

File: README.md [hunk 7]
```diff
@@ -43,12 +43,12 @@ Get started with the benchmarks:
...
-| [Tailcall]       |   `2,890.68` |      `34.69` |
-| [Gqlgen]         |     `935.00` |     `115.73` |
-| [Apollo GraphQL] |     `793.37` |     `128.22` |
-| [Netflix DGS]    |     `597.39` |     `191.85` |
+| Server | Requests/sec | Latency (ms) |
+|--------:|--------------:|--------------:|
+| [Tailcall] | `6066.06` | `16.46` |
+| [Gqlgen] | `2088.87` | `49.66` |
+| [Apollo GraphQL] | `1673.16` | `59.97` |
+| [Netflix DGS] | `1444.94` | `73.29` |
```
Comment (ologbonowiwi) [inline README.md:51]: I've pushed the changes to use `,` as a separator, but the CI is still running https://github.com/ologbonowiwi/graphql-benchmarks/actions/runs/6902336459/job/18778839441

**Review by tusharmath — APPROVED** [review]

**Pull request merged**